iree/compiler/Dialect/LinalgPlus/IR/LinalgPlusOps.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/LinalgPlus/IR/LinalgPlusOps.h"
+
+#include "iree/compiler/Dialect/LinalgPlus/IR/LinalgPlusDialect.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/Support/SMLoc.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/Diagnostics.h"
+#include "mlir/IR/OpImplementation.h"
+#include "mlir/IR/OperationSupport.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "mlir/IR/Value.h"
+#include "mlir/Support/LLVM.h"
+#include "mlir/Support/LogicalResult.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace linalg_plus {
+
+void SortOp::build(OpBuilder &builder, OperationState &state,
+                   ValueRange operands, int64_t dimension) {
+  state.addOperands(operands);
+  state.addAttribute("dimension", builder.getI64IntegerAttr(dimension));
+
+  for (Value operand : operands) state.addTypes(operand.getType());
+
+  state.addRegion();
+}
+
+static LogicalResult verifySortOp(SortOp op) {
+  Operation::operand_range operands = op.operands();
+  if (operands.empty()) return op.emitOpError("requires at least one input");
+
+  if (llvm::all_of(operands, [](Value operand) {
+        return operand.getType().cast<ShapedType>().hasRank();
+      })) {
+    ArrayRef<int64_t> inputShape =
+        (*operands.begin()).getType().cast<ShapedType>().getShape();
+
+    if (llvm::any_of(llvm::drop_begin(operands, 1), [&](Value operand) {
```

Comment:
Why is this nested. This n^2 operations.

Reply:
This was copied from the implementation of mhlo.sort.